cmd/mindthegap/flags/custom_flags.go

```diff
@@ -0,0 +1,56 @@
+package flags
+
+import (
+	"strings"
+)
+
+const (
+	httpScheme  = "http"
+	httpsScheme = "https"
+
+	httpSchemePrefix  = "http://"
+	httpsSchemePrefix = "https://"
+)
+
+type RegistryURI struct {
+	raw     string
+	scheme  string
+	address string
+}
+
+func (v RegistryURI) String() string {
+	return v.raw
+}
+
+func (v *RegistryURI) Set(value string) error {
+	v.raw = value
+	v.scheme, v.address = parsePossibleURI(value)
+
+	return nil
+}
+
+func parsePossibleURI(value string) (string, string) {
```

Comment:
The way this parsing works smells to me, but `url.Parse` was not very helpful here
```
parse "0.0.0.0:5000": first path segment in URL cannot contain colon
```

Reply:
It was a little tricky because for example a totally valid registry `0.0.0.0:5000/dkp` fails parsing `url.Parse()/ParseRequestURI` see [here](https://go.dev/play/p/qX53W9OVBe1).

Found some inspiration from https://stackoverflow.com/a/62083424